tla-pp/src/main/scala/at/forsyte/apalache/tla/pp/SetMembershipSimplifier.scala

```diff
@@ -25,19 +25,41 @@ class SetMembershipSimplifier(tracker: TransformationTracker) extends AbstractTr
     transform(expr)
   }
 
-  private def typeOfPredefSet: PartialFunction[TlaPredefSet, TlaType1] = {
+  /**
+   * Returns the type of a TLA+ predefined set, if this transformation is applicable. In particular, it is *not*
+   * applicable to `Nat`, since `i \in Nat` does not hold for all `IntT1`-typed `i`.
+   */
+  private def typeOfSupportedPredefSet: PartialFunction[TlaPredefSet, TlaType1] = {
     case TlaBoolSet => BoolT1()
     case TlaIntSet  => IntT1()
     case TlaRealSet => RealT1()
     case TlaStrSet  => StrT1()
+    // intentionally omits TlaNatSet, see above.
   }
```

Comment:
I'm afraid I was overly hasty and did a bad job of getting up speed on this PR and the proposed changes after being tagged in. I had missed that this function was necessary as an auxiliary, and was replying under the misconception that we could avoid having to manually enumerate this mapping. It was the prospect of avoiding this manual enumeration that mainly motivated my reasoning in my replies. If we still have to enumerate this mapping, but we just move it into a less direct location, I agree we don't get really get a beneficial abstraction or improved readability. My bad. 

I'm sorry for the misunderstanding, and I'll be more careful in my reviews/opinions in the future.

Reply:
@shonfeder No worries, I also missed it at first! Luckily my brain kept processing overnight 😅